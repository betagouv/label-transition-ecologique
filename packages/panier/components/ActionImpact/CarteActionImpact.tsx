import classNames from 'classnames';
import {CarteActionImpactProps} from './types';
import {Badge, Button, Card} from '@tet/ui';
import NiveauBudget from './NiveauBudget';

/**
 * Carte action à impact du panier d'actions
 */

export const CarteActionImpact = ({
  titre,
  thematiques,
  budget,
  panier,
  isSelected,
  statut,
  onToggleSelected,
  onUpdateStatus,
}: CarteActionImpactProps) => {
  const handleToggleSelect = (value: boolean) => onToggleSelected(value);

  return (
    <Card
      className={classNames('box-content px-4 py-4 h-28', {
        '!cursor-default': panier && !isSelected,
      })}
      isSelected={isSelected}
      footer={
        <div className="relative z-0">
          <div className="flex justify-end items-center gap-2 opacity-100 group-hover:opacity-0 transition-opacity duration-500">
            {/* Badge thématique */}
            {!!thematiques.length && (
              <Badge title={thematiques[0].nom} size="sm" state="standard" />
            )}
            {/* Budget */}
            <NiveauBudget budget={budget ?? {niveau: 4, nom: 'Non estimé'}} />
          </div>

          {/* Boutons d'action, visibles au hover de la carte */}
          <div className="opacity-0 group-hover:opacity-100 transition-opacity duration-500 w-full absolute z-10 bottom-0 right-0 bg-white flex justify-end gap-2">
            {panier ? (
              <Button size="xs" onClick={() => handleToggleSelect(false)}>
                Retirer du panier
              </Button>
            ) : (
              <>
                <Button
                  variant="outlined"
                  size="xs"
                  className={classNames({
                    '!bg-primary-2': statut === 'en_cours',
                  })}
                  onClick={() => onUpdateStatus?.('en_cours')}
                >
                  En cours
                </Button>
                <Button
                  variant="outlined"
                  size="xs"
                  className={classNames({
                    '!bg-primary-2': statut === 'realise',
                  })}
                  onClick={() => onUpdateStatus?.('realise')}
                >
                  Réalisée
                </Button>

                <Button size="xs" onClick={() => handleToggleSelect(true)}>
                  Ajouter au panier
                </Button>
              </>
            )}
          </div>
        </div>
      }
    >
      <span className="line-clamp-3">{titre}</span>
    </Card>
  );
};
