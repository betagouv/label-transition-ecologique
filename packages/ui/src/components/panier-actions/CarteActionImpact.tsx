import {Badge} from '@design-system/Badge';
import {Button} from '@design-system/Button';
import {Card} from '@design-system/Card';
import classNames from 'classnames';
import {valeurToBadge} from './utils';
import {NiveauBudget} from './NiveauBudget';
import {CarteActionImpactProps} from './types';

const maxLength = 130;

const splitTitle = (title: string) => {
  let newContent = title.slice(0, maxLength);
  const contentEnd = title.slice(maxLength).split(' ')[0];
  newContent += contentEnd;
  return newContent;
};

/**
 * Carte action à impact du panier d'actions
 */

export const CarteActionImpact = ({
  titre,
  thematiques,
  complexite,
  budget,
  panier,
  isSelected,
  statut,
  onToggleSelected,
  onUpdateStatus,
}: CarteActionImpactProps) => {
  const handleToggleSelect = value => onToggleSelected(value);

  return (
    <Card
      className="box-content !px-5 !py-4 h-[160px]"
      isSelected={isSelected}
      header={
        <div className="flex justify-between">
          {/* Catégorie */}
          <div>{thematiques[0]?.nom ?? ''}</div>

          {/* Budget */}
          <NiveauBudget budget={budget} />
        </div>
      }
      footer={
        <div className="relative z-0">
          {/* Badge de complexité */}
          <Badge
            title={`Complexité ${valeurToBadge[complexite].nom}`}
            size="sm"
            state={valeurToBadge[complexite].style}
          />

          {/* Boutons d'action, visibles au hover de la carte */}
          <div className="invisible group-hover:visible w-full absolute z-10 bottom-0 right-0 bg-white flex justify-end gap-2">
            {panier ? (
              <Button size="xs" onClick={() => handleToggleSelect(false)}>
                Retirer du panier
              </Button>
            ) : (
              <>
                <Button
                  variant="outlined"
                  size="xs"
                  className={classNames({
                    '!bg-primary-2': statut === 'en_cours',
                  })}
                  onClick={() => onUpdateStatus?.('en_cours')}
                >
                  En cours
                </Button>
                <Button
                  variant="outlined"
                  size="xs"
                  className={classNames({
                    '!bg-primary-2': statut === 'realise',
                  })}
                  onClick={() => onUpdateStatus?.('realise')}
                >
                  Réalisée
                </Button>

                <Button size="xs" onClick={() => handleToggleSelect(true)}>
                  Ajouter
                </Button>
              </>
            )}
          </div>
        </div>
      }
    >
      {titre.length > maxLength ? `${splitTitle(titre)}...` : titre}
    </Card>
  );
};
