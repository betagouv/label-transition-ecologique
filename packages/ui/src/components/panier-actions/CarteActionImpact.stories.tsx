import {Meta, StoryObj} from '@storybook/react';
import {CarteActionImpact} from './CarteActionImpact';

const meta: Meta<typeof CarteActionImpact> = {
  title: "Composants métier/Panier d'actions/Carte action à impact",
  component: CarteActionImpact,
  argTypes: {
    complexite: {
      control: {type: 'select'},
    },
    budget: {
      control: {type: 'select'},
    },
  },
  args: {
    titre:
      'Désigner une (ou plusieurs personnes) chef de projet transition écologique',
    categorie: 'Mieux consommer',
    complexite: 1,
    budget: 2,
    onToggleSelected: () => {},
    onUpdateStatus: () => {},
  },
};

export default meta;

type Story = StoryObj<typeof CarteActionImpact>;

export const Default: Story = {};

export const CarteSelectionnee: Story = {
  render: args => <CarteActionImpact {...args} isSelected={true} />,
};
