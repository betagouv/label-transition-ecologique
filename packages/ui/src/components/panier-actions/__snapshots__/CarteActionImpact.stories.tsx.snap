// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Panier d'actions/Carte action à impact CarteSelectionnee smoke-test 1`] = `
<div class="p-7 m-px bg-white rounded-lg flex flex-col gap-4 text-primary-9 text-base font-bold group hover:m-0 hover:cursor-pointer hover:shadow-md !m-0 border-2 border-info-1 box-content !px-5 !py-4 h-[160px]">
  <div class="text-primary-8 text-sm font-bold">
    <div class="flex justify-between">
      <div>
        Mieux consommer
      </div>
      <div>
        <span class="ri-money-euro-circle-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-secondary-1">
        </span>
        <span class="ri-money-euro-circle-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-secondary-1">
        </span>
        <span class="ri-money-euro-circle-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-grey-4">
        </span>
      </div>
    </div>
  </div>
  Désigner une (ou plusieurs personnes) chef de projet transition écologique
  <div class="text-primary-8 text-sm font-bold mt-auto">
    <div class="relative z-0">
      <div data-test="Badge-undefined"
           class="bg-success-2 border-success-2 flex items-center gap-1 max-w-max h-fit px-3 py-1 border border-solid rounded !px-1.5 !py-0.5"
      >
        <span class="text-success-1 font-bold uppercase leading-4 text-left line-clamp-1 text-xs">
          Complexité simple
        </span>
      </div>
      <div class="invisible group-hover:visible w-full absolute z-10 bottom-0 right-0 bg-white flex justify-end gap-2">
        <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !px-3">
          Non pertinente
        </button>
        <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !px-3">
          En cours
        </button>
        <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !px-3">
          Réalisée
        </button>
        <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !px-2">
          Ajouter
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components/Panier d'actions/Carte action à impact Default smoke-test 1`] = `
<div class="p-7 m-px bg-white rounded-lg flex flex-col gap-4 text-primary-9 text-base font-bold group border border-primary-3 box-content !px-5 !py-4 h-[160px]">
  <div class="text-primary-8 text-sm font-bold">
    <div class="flex justify-between">
      <div>
        Mieux consommer
      </div>
      <div>
        <span class="ri-money-euro-circle-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-secondary-1">
        </span>
        <span class="ri-money-euro-circle-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-secondary-1">
        </span>
        <span class="ri-money-euro-circle-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] text-grey-4">
        </span>
      </div>
    </div>
  </div>
  Désigner une (ou plusieurs personnes) chef de projet transition écologique
  <div class="text-primary-8 text-sm font-bold mt-auto">
    <div class="relative z-0">
      <div data-test="Badge-undefined"
           class="bg-success-2 border-success-2 flex items-center gap-1 max-w-max h-fit px-3 py-1 border border-solid rounded !px-1.5 !py-0.5"
      >
        <span class="text-success-1 font-bold uppercase leading-4 text-left line-clamp-1 text-xs">
          Complexité simple
        </span>
      </div>
      <div class="invisible group-hover:visible w-full absolute z-10 bottom-0 right-0 bg-white flex justify-end gap-2">
        <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !px-3">
          Non pertinente
        </button>
        <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !px-3">
          En cours
        </button>
        <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !px-3">
          Réalisée
        </button>
        <button class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !px-2">
          Ajouter
        </button>
      </div>
    </div>
  </div>
</div>
`;
